srcStatic/IniFile.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <string>
+
+
+class IniSection;
+
+
+// Represents an entire .ini file
+class IniFile {
+public:
+
+	IniFile(std::string fileName);
+
+	const std::string& FileName() const;
```

Comment:
I think this makes returning the filename dependent on not deconstructing the class since the saved instance of the string referenced is contained within the class?

This is probably an okay assumption to make, and I assume it provides a bit of optimization by not having to copy the filename when referenced. I'm not sure the 'risk' is worth the reward here considering the filename will likely not be referenced repeatedly in a performance hampering way. Although my thoughts on the matter I think are more of an opinion than anything else.

Same would apply to a couple of GET functions from the IniSection class.

Reply:
Yes, you're right. It ties the lifetime of the result value to the lifetime of the object. Yes, that can open up the possibility of a dangling reference, which the compiler is not likely to warn about.

The likelihood of code being written that triggers that behaviour seems small, and is easy enough to work around. I figured that solution was safe enough here since the uses are likely for immediate consumption in logging or display purposes. By returning a reference it can be done more efficiently, and by making it `const`, the internals of the class are still protected.

If the value needs to have its lifetime extended, it can always be copied to a new string when capturing the value. Example:
```cpp
std::string name = iniFile.FileName(); // Copy to new std::string
```

That way the calling code can decide if it wants efficiency, or generality. I figured it went along with the C/C++ ethos of not paying for things you don't need.

Also, this conversation very much applies to how we want to approach using `std::string_view` in the future. It's the exact same concept if you return a non-owning read-only `std::string_view` from a function. Although you need to beware, I think in some cases it's worth it. Subjective argument though of course.